ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/KsqlResourceTest.java

```diff
@@ -1932,6 +1932,21 @@ public void shouldFailAllCommandsIfWouldReachActivePersistentQueriesLimit() {
     verify(commandStore, never()).enqueueCommand(any(), any(), any());
   }
 
+  @Test
+  public void shouldNotFailIfNotQueryDespiteReachedActivePersistentQueriesLimit() {
+    // Given:
+    givenKsqlConfigWith(
+        ImmutableMap.of(KsqlConfig.KSQL_ACTIVE_PERSISTENT_QUERY_LIMIT_CONFIG, 3));
+
+    givenMockEngine();
+
+    // mock 3 queries already running + 1 new query to execute
+    givenPersistentQueryCount(4);
```

Comment:
If the limit is 3, then we should have already gotten an exception for the 4th persistent query, right?
I think this test could check that if we have 3 persistent queries and we issue a `SHOW STREAMS`, then no exception is thrown. But if we issue another persistent query, then an exception is thrown

Reply:
`givenPersistentQueryCount` does not issue 4 queries, it just mocks a function that counts persistent queries. I actually ended up setting the limit to 6, so that it replicates the scenario we saw which is a cluster being restarted with `KSQL_ACTIVE_PERSISTENT_QUERY_LIMIT_CONFIG` set to a very low number.

Reply:
+1 to @vpapavas's question, it looks like this test only works because it's already greater than the configured allowed number of queries. perhaps that's the change we want to make (because there are situations where we may accidentally reduce the number of persistent queries allowed on a cluster after it's already exceeded) but we should make that clear

Reply:
So, the scenario you are trying to emulate is that `KSQL_ACTIVE_PERSISTENT_QUERY_LIMIT_CONFIG` was set to a number, let's say 4. Then, a cluster is getting restarted and that config changes to let's say 2. In this case, no further persistent queries should be able to get created but `SHOW STREAMS` should still succeed. If the users try to issue another persistent query, they should get the exception. Correct? Can you change the test, to represent this behavior and also change the name and add a description?

Reply:
Updated the test to make the behaviour we're testing more clear.